File: army/JavaArmy/src/com/maksklaban/armymax/state/WolfState.java
package com.maksklaban.armymax.state;

import com.maksklaban.armymax.exceptions.*;

public class WolfState extends State {
    public WolfState(int hitPoints, int damage, String title) {
        super(hitPoints + (hitPoints / 2), damage * 2, title);
    }

    public WolfState(int hitPoints, int damage) {
        this(hitPoints, damage, "WolfForm");
    }

    public void takeMagDamage(int hp) throws ArmyException {
        super.takeMagDamage(hp*2);
    }

}

File: army/State/WolfState.cpp
#include "WolfState.h"

WolfState::WolfState(int hitPoints, int damage, std::string title) : State(hitPoints * 1.5, damage * 2, title) {}

WolfState::~WolfState() {}

void WolfState::takeMagDamage(int dmg) {
    State::takeMagDamage(dmg*2);
}

